src/lint/validators.nim

```diff
@@ -20,14 +20,17 @@ func format(context, key: string): string =
   else:
     q(key)
 
-proc hasKey(data: JsonNode; key, path: string; isRequired: bool;
-            context = ""): bool =
-  ## Returns true if `data` contains the key `key`. Otherwise, returns false
-  ## and, if `isRequired` is true, prints an error message.
+template handleMissingKey(data: JsonNode; key, path, context: string;
+                          isRequired: bool; body: untyped) =
+  ## Runs `body` if `data` contains the key `key`. Otherwise: prints an error
+  ## message if `isRequired` is true, and sets `result` to true if `isRequired`
+  ## is false.
   if data.hasKey(key):
-    result = true
+    body
   elif isRequired:
-    result.setFalseAndPrint(&"Missing key: {format(context, key)}", path)
+    result.setFalseAndPrint("Missing key: " & format(context, key), path)
```

Comment:
This change is required because we can't use strformat like that inside a template (due to strformat implementation details).